Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -378,10 +378,26 @@ def start_upgrade
 
       disable_accessibility_access
       uninstall_artifacts
+      backup if version_is_latest?
```

Comment:
Let's always create a backup, not just when `version_is_latest?`.

Reply:
If we always create a backup, we'll need a way to know if this is the new or old cask when `purge_versioned_files`. Otherwise, it will leave a dangling, empty folder if the upgrade fails (see the test `"restores the old Cask if the upgrade failed"` in `upgrade_spec.rb`)

Reply:
Isn't `start_upgrade` only called on the old cask?

Reply:
Yes, I took advantage of that; see next commit.